src/material-experimental/mdc-slider/slider.ts

```diff
@@ -977,7 +977,28 @@ class SliderAdapter implements MDCSliderAdapter {
     return this._delegate._getInputElement(thumbPosition).getAttribute(attribute);
   }
   setInputAttribute = (attribute: string, value: string, thumbPosition: Thumb): void => {
-    this._delegate._getInputElement(thumbPosition).setAttribute(attribute, value);
+    const input = this._delegate._getInputElement(thumbPosition);
+
+    // Explicitly check the attribute we are setting to prevent xss.
+    switch (attribute) {
+      case 'aria-valuetext':
+        input.setAttribute('aria-valuetext', value);
+        break;
+      case 'disabled':
+        input.setAttribute('disabled', value);
+        break;
+      case 'min':
+        input.setAttribute('min', value);
+        break;
+      case 'max':
+        input.setAttribute('max', value);
+        break;
+      case 'value':
+        input.setAttribute('value', value);
+        break;
+      case 'step':
+        input.setAttribute('step', value);
+    }
```

Comment:
I'm a bit rusted on XSS but in what cases would an attacker be able to have `setInputAttribute` invoked with an arbitrary name and value through XSS?

Also if we'd go with that; I think we should have a default case and throw an error. This would notify us in case MDC starts calling this adapter method with a different attribute we would want to handle too (and not skip silently).

Reply:
It would be tough to pull an xss like this off but theoretically this would be the setup for the attack:

1. The MDC Foundation starts setting a vulnerable attribute (this is a stupid example, but lets say they're setting the `onclick` attribute).
2. The MDC Foundation is getting the value for the attribute from the end developer.
3. The end developer is using a value set by the user (in this example, we could use `'alert("xss")'` as the value).

The result would be us calling `setAttribute('onclick', 'alert("xss")')`.

This is obviously a very unrealistic example, but internally there is a check to ensure we don't call `setAttribute` without knowing the attribute we're setting. A more realistic example of this would be if the mdc team is using `setAttribute` to set the `href` of a link button.

Idk why this error wasn't getting thrown for our other components. I have a few ideas why but haven't looked into it.

Reply:
Can't there always be cases like that? I think the actual XSS protection should be performed by the MDC foundation that calls this adapter method. As of right now their foundation seems to not pass any user values to the adapter method.

e.g. there are places in FW too where `setAttribute` is "proxied", but ultimately it's up to the consumer to not pass any user value to it that could allow for an XSS. Do I miss something? has this been flagged as a potential XSS source in g3?

Reply:
resolved offline. we're adding a todo